src/notebooks/notebookCommandListener.ts

```diff
@@ -175,6 +187,20 @@ export class NotebookCommandListener implements IDataScienceCommandListener {
         this.runAllCells();
     }
 
+    private async restartKernelAndRunUpToSelectedCell(notebookUri: Uri | undefined) {
+        const activeNBE = this.notebookEditorProvider.activeNotebookEditor;
```

Comment:
do we need `this.notebookEditorProvider.activeNotebookEditor` here? It's possible that restart happens on an inactive notebook editor but this check will block the restart from running.

Reply:
Fair point, then I'll shift to taking the uri as done on L#193 and then it should be covered either case. Just need to change the range on L#197